cohort-engine/src/main/java/com/ibm/cohort/engine/measure/R4ParameterDefinitionWithDefaultToCohortParameterConverter.java

```diff
@@ -0,0 +1,149 @@
+package com.ibm.cohort.engine.measure;
+
+import java.util.stream.Collectors;
+
+import org.hl7.fhir.r4.model.Base64BinaryType;
+import org.hl7.fhir.r4.model.BooleanType;
+import org.hl7.fhir.r4.model.CodeableConcept;
+import org.hl7.fhir.r4.model.Coding;
+import org.hl7.fhir.r4.model.DateTimeType;
+import org.hl7.fhir.r4.model.DateType;
+import org.hl7.fhir.r4.model.DecimalType;
+import org.hl7.fhir.r4.model.Extension;
+import org.hl7.fhir.r4.model.InstantType;
+import org.hl7.fhir.r4.model.IntegerType;
+import org.hl7.fhir.r4.model.ParameterDefinition;
+import org.hl7.fhir.r4.model.Period;
+import org.hl7.fhir.r4.model.Quantity;
+import org.hl7.fhir.r4.model.Range;
+import org.hl7.fhir.r4.model.Ratio;
+import org.hl7.fhir.r4.model.StringType;
+import org.hl7.fhir.r4.model.TimeType;
+import org.hl7.fhir.r4.model.Type;
+import org.hl7.fhir.r4.model.UriType;
+
+import com.ibm.cohort.engine.cdm.CDMConstants;
+import com.ibm.cohort.engine.measure.parameter.UnsupportedFhirTypeException;
+import com.ibm.cohort.engine.parameter.BooleanParameter;
+import com.ibm.cohort.engine.parameter.CodeParameter;
+import com.ibm.cohort.engine.parameter.ConceptParameter;
+import com.ibm.cohort.engine.parameter.DateParameter;
+import com.ibm.cohort.engine.parameter.DatetimeParameter;
+import com.ibm.cohort.engine.parameter.DecimalParameter;
+import com.ibm.cohort.engine.parameter.IntegerParameter;
+import com.ibm.cohort.engine.parameter.IntervalParameter;
+import com.ibm.cohort.engine.parameter.Parameter;
+import com.ibm.cohort.engine.parameter.QuantityParameter;
+import com.ibm.cohort.engine.parameter.RatioParameter;
+import com.ibm.cohort.engine.parameter.StringParameter;
+import com.ibm.cohort.engine.parameter.TimeParameter;
+
+/**
+ * Expected types to handle derived from this definition:
+ * https://pages.github.ibm.com/watson-health-fhir-server/ig-common-data-model/StructureDefinition-parameter-definition-with-default.html
+ */
+public class R4ParameterDefinitionWithDefaultToCohortParameterConverter {
+
+	private R4ParameterDefinitionWithDefaultToCohortParameterConverter() {
+	}
+
+	public static Parameter toCohortParameter(ParameterDefinition parameterDefinition) {
+		Extension defaultValueExtension = parameterDefinition.getExtensionByUrl(CDMConstants.PARAMETER_DEFAULT_URL);
+
+		Parameter parameter = null;
+		if (defaultValueExtension != null) {
```

Comment:
90% of this method operates on the Extension and only this little bit at the top operates on the ParameterDefinition. Purely as a matter of code style, I would extract the part that deals with just the extension into a separate method ``toCohortParameter(Extension extension)`` and call it from here.

Reply:
Would you want `toCohortParameter(Extension extension)` to be `public`, or are you fine leaving the primary public API requiring a  ParameterDefinition and having the extracted method stay `private`?

Reply:
I would make it public. I have this idea in my head that we might want to convert and extension to a CQL parameter somewhere else and it might be useful.